### DISCUSSION TITLE
Colors differ between Storybook and Docs

### DISCUSSION THREAD

**Discussion by FoundationKen** [rewindui/rewindui#116] [General]
I got Rewind UI working in my app, but I noticed that the colors differ from what is shown on the documentation pages.  I looked into it and noticed that the online storybook has the same colors as my app, so it is also different from the docs.

Compare these two accordions, for example:

Storybook: https://storybook.rewind-ui.dev/?path=/docs/components-accordion--docs

Docs: https://rewind-ui.dev/components/accordion

The style `.bg-purple-500` is defined differently in each case.

What is the source of difference between these sets of colors, and how can I use the same colors as in the docs (I like that palette more)?

Thanks!

**Comment by NickDunas** [comment]
Hello there!

Yes that's on purpose. I just wanted to use a different color palette on the docs page instead of the default tailwind colors.

You can find the docs configuration here: https://github.com/rewindui/rewindui/blob/main/apps/docs/tailwind.config.js

**Reply by FoundationKen** [reply]
You are so fast!  I actually just found it on my own and got it working!  Thank you!

One other color issue is that my Alert  background in `tone="light"` mode is white instead of light purple.  The `bg-purple-100/95` is not seen by Tailwind for some reason.  If I add the same style directly to another component by `className="bg-purple-100/95"` then the style is also applied to the Alert, but I can't see why the style is not seen on its own.

My `tailwind.config.js` has `  content: ['./src/**/*.{html,jsx,tsx}', './node_modules/@rewind-ui/core/dist/theme/styles/*.js'],` and that is definitely being parsed, because removing it messes everything up.

<img width="773" alt="image" src="https://github.com/user-attachments/assets/560edd50-dd5a-4ec8-8f42-748b0de5958d">

Any ideas what I am missing?

**Reply by NickDunas** [reply]
Are you using vite?

**Reply by FoundationKen** [reply]
Yes.  Maybe something to do with `class-variance-authority`.  Adding the styles to the safelist makes it work, but that is not ideal.

**Reply by FoundationKen** [reply]
This is my `vite.config.js`:
```js
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react-swc'
import { TanStackRouterVite } from '@tanstack/router-plugin/vite'
import tailwindcss from 'tailwindcss'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [TanStackRouterVite(), react()],
  css: {
    postcss: {
      plugins: [tailwindcss()],
    },
  },
  build: {
    outDir: 'build',
  },
  server: {
    port: 5173, // Or any other port you prefer
    host: true,
  },
})
```

And this is my `tailwind.config.js`:

```js
import typography from '@tailwindcss/typography'
import scrollbar from 'tailwind-scrollbar'
import forms from '@tailwindcss/forms'
import colors from 'tailwindcss/colors'

/** @type {import('tailwindcss').Config} */
export default {
  mode: 'jit',
  content: ['./src/**/*.{html,jsx,tsx}', './node_modules/@rewind-ui/core/dist/theme/styles/*.js'],
  plugins: [
    typography,
    scrollbar({ nocompatible: true }),
    forms({ strategy: 'class' }), // only generate classes
  ],
}
```

**Reply by NickDunas** [reply]
For some reason tailwind cannot when using vite cannot find some classes properly. You might notice that classes like m-1.5 (classes with decimals), are missing as well. My workaround on this, is to copy the whole style.ts file from the rewindui repo and add it on my own project and add this folder on my tailwind config as well. 

I have really tried multiple times to figure out the reason tailwind is missing these classes when used with vite, but I did not manage to find a proper solution.

**Reply by FoundationKen** [reply]
Ah, I see.  Can you link the file?  I can't find it.
This could be a work around for now.

**Reply by FoundationKen** [reply]
Maybe we can come up with a minimal list of patterns and use the `safelist`, like this (just for demonstration):

```js
safelist: [
    // Add specific classes
    { pattern: /bg-.*\/95/ },
  ],
```

**Reply by NickDunas** [reply]
This might work as well. You can find all style files here: https://github.com/rewindui/rewindui/tree/main/packages/core/src/theme/styles

**Reply by FoundationKen** [reply]
I think this is related to the `*.styles.js` files all being on one line.  If I replace `Alert.styles.js` in my node modules with the version from the repo, then the styles work.

Can you modify the build so that these files are not crunched down to a single line?

Tailwind may have a max line length that it can handle when looking for styles in strings.

**Reply by FoundationKen** [reply]
This change to the `rollup.config.js` fixes the issue:

```js
import commonjs from '@rollup/plugin-commonjs';
import resolve from '@rollup/plugin-node-resolve';
import typescript from '@rollup/plugin-typescript';
import terser from '@rollup/plugin-terser';
import { exec } from 'child_process';

import copy from 'rollup-plugin-copy';
import peerDepsExternal from 'rollup-plugin-peer-deps-external';
import { typescriptPaths } from 'rollup-plugin-typescript-paths';
import preserveDirectives from 'rollup-plugin-preserve-directives';

const outputOptions = {
  sourcemap: false,
  preserveModules: true,
  preserveModulesRoot: 'src',
};

const tscAlias = () => {
  return {
    name: 'tsAlias',
    writeBundle: () => {
      return new Promise((resolve, reject) => {
        exec('tsc-alias', function callback(error, stdout, stderr) {
          if (stderr || error) {
            reject(stderr || error);
          } else {
            resolve(stdout);
          }
        });
      });
    },
  };
};

// Custom plugin to conditionally apply the terser plugin
const conditionalTerser = () => {
  return {
    name: 'conditional-terser',
    renderChunk(code, chunk, options) {
      const excludedPatterns = [/theme\/styles\/.*.styles.js/];

      if (excludedPatterns.some((pattern) => pattern.test(chunk.fileName))) {
        return { code, map: null }; // Skip minification
      }

      // Apply terser if the file is not excluded
      const result = terser().renderChunk.call(this, code, chunk, {
        compress: { directives: false },
      });
      return result;
    },
  };
};

const config = [
  {
    input: 'src/index.ts',
    output: [
      {
        dir: 'dist',
        format: 'cjs',
        entryFileNames: '[name].cjs',
        exports: 'auto',
        ...outputOptions,
      },
      {
        dir: 'dist',
        format: 'esm',
        ...outputOptions,
      },
    ],
    external: [
      'react',
      'react-dom',
      '@babel/runtime',
      '@floating-ui/react',
      '@floating-ui/react-dom',
      'tailwind-merge',
      'tslib',
      'react-focus-lock',
      'class-variance-authority',
    ],
    plugins: [
      peerDepsExternal(),
      resolve(),
      commonjs(),
      typescript({
        tsconfig: './tsconfig.json',
        exclude: ['**/stories/**', '**/tests/**', './styles.css'],
      }),
      typescriptPaths(),
      preserveDirectives(),
      conditionalTerser(),
      copy({
        targets: [{ src: './../../README.md', dest: 'dist' }],
      }),
      tscAlias(),
    ],
    onwarn(warning, warn) {
      if (warning.code !== 'MODULE_LEVEL_DIRECTIVE') {
        warn(warning);
      }
    },
  },
];

export default config;
```

Let me know what you think of this approach, and I can open a PR if you want to finalize/improve it.

**Reply by NickDunas** [reply]
Oh I really really thank you for this. It has been bugging me for so long!

Yes please open a PR and I will merge it as soon as possible!

**Reply by FoundationKen** [reply]
PR sent from my other account (where I am actually using RewindUI).

https://github.com/rewindui/rewindui/pull/117

**Reply by NickDunas** [reply]
I just merged your PR and published a new version. Please give it a test if you want and let me know if it works as expected now!

**Comment by FoundationKen** [comment]
Yes!  It works now with no safelist.

**Reply by NickDunas** [reply]
I really thank you! 🙏
